components/eamxx/src/physics/p3/impl/p3_init_impl.hpp

```diff
@@ -3,35 +3,327 @@
 
 #include "p3_functions.hpp" // for ETI only but harmless for GPU
 
-extern "C" {
-  void micro_p3_utils_init_c(scream::Real Cpair, scream::Real Rair, scream::Real RH2O, scream::Real RHO_H2O,
-                 scream::Real MWH2O, scream::Real MWdry, scream::Real gravit, scream::Real LatVap, scream::Real LatIce,
-                 scream::Real CpLiq, scream::Real Tmelt, scream::Real Pi, bool masterproc);
-  void p3_init_c(const char** lookup_file_dir, int* info, const bool& write_tables);
-}
+#include "ekat/util/ekat_file_utils.hpp"
+
+#include <fstream>
 
 namespace scream {
 namespace p3 {
 
+namespace {
+
+template <typename S, typename IceT, typename CollT>
+void read_ice_lookup_tables(const bool masterproc, const char* p3_lookup_base, const char* p3_version, IceT& ice_table_vals, CollT& collect_table_vals, int densize, int rimsize, int isize, int rcollsize)
+{
+  using DeviceIcetable = typename IceT::non_const_type;
+  using DeviceColtable = typename CollT::non_const_type;
+
+  const auto ice_table_vals_d     = DeviceIcetable("ice_table_vals");
+  const auto collect_table_vals_d = DeviceColtable("collect_table_vals");
+
+  const auto ice_table_vals_h    = Kokkos::create_mirror_view(ice_table_vals_d);
+  const auto collect_table_vals_h = Kokkos::create_mirror_view(collect_table_vals_d);
+
+  //
+  // read in ice microphysics table into host views. We always read these as doubles.
+  //
+
+  std::string filename = std::string(p3_lookup_base) + std::string(p3_version);
+
+  if (masterproc) {
+    std::cout << "Reading ice lookup tables in file: " << filename << std::endl;
+  }
+
+  std::ifstream in(filename);
```

Comment:
Does this mean that the tables are plain ascii now? I kind of like that, so one can always inspect the file content manually, and since they are relatively small and only used at init time, binary format is not that important.

Reply:
@bartgol , I should have made it more clear in the PR description that this function impl is, for the most part, not new. It's just what `init_kokkos_ice_lookup_tables` was before except moved here and renamed.